wireshark/README.md

```diff
@@ -1,31 +1,81 @@
+          ```
+          $env:DISPLAY="<your_host_ip>:0"
+          ```
+    - **macOS**:
+      - Install an X server, such as [XQuartz](https://www.xquartz.org), and start it.
+      - Run the following command in the terminal to allow Docker to access the display:
+        ```
+        xhost +
+        ```
+      - Set the display environment variable in the terminal:
+        ```
+        export DISPLAY=<your_host_ip>:0
+        ```
 
-## Quick start
+**Notes**:
 
-1) Be sure to have [docker installed](https://docs.docker.com/install).
-2) Run `docker build -t openquantumsafe/wireshark .` to create an QSC-enabled (codepoint and OID aware) wireshark docker image.
+- **macOS** support has not been tested yet. We welcome your feedback and suggestions. Please reach us through
+  the [oqs-demos issue section](https://github.com/open-quantum-safe/oqs-demos/issues).
+- Replace `<your_host_ip>` with your system's IP address. Use `:0` as the default display port unless configured
+  otherwise.
 
-## Usage
+## Building Instructions
+
+Run the following commands to build and launch Wireshark with OQS support:
+
+```
+git clone https://github.com/open-quantum-safe/oqs-demos
+cd oqs-demos/wireshark
+docker build -t oqs-wireshark .
+docker run --rm -it --net=host -e DISPLAY=$DISPLAY -v /tmp/.X11-unix:/tmp/.X11-unix oqs-wireshark
+```
 
-Information how to use the image is [available in the separate file USAGE.md](USAGE.md).
+### Explanation of Docker Options
 
-## Build options
+- `--net=host`: Shares the host network with the container.
+- `-e DISPLAY`: Sets the display variable for GUI.
+- `-v /tmp/.X11-unix:/tmp/.X11-unix`: Mounts the X11 Unix socket for GUI access.
 
-The Dockerfile provided allows for customization of the image built:
+## Project Components
 
-### WIRESHARK_VERSION
+1. **Dockerfile**: Builds Wireshark with OpenSSL, liboqs, and OQS provider.
+2. **generate_qsc_header.py**: Processes `oqs-provider/oqs-template/generate.yml` with the `qsc_template.jinja2` to
+   generate `qsc.h`,
+   defining post-quantum KEMs and SIGs for Wireshark.
+
+## Usage
 
-This permits changing the wireshark code base to be used. 
+For detailed usage instructions, refer to [USAGE.md](USAGE.md).
 
-Tested default value is "3.4.9".
+## Build Configuration and Updates
 
-### QSC_SSL_FLAVOR
+Customize the build using the following Dockerfile arguments:
 
-Different quantum-safe TLS implementations have different names for the same algorithms. This option permits switching between them. Permitted values are "oqs" and "wolfssl".
+- **`UBUNTU_VERSION`**: Specifies the Ubuntu version.
+- **`WIRESHARK_VERSION`**: Defines the Wireshark version to build.
+- **`OPENSSL_TAG`**: Sets the OpenSSL version to build.
+- **`LIBOQS_TAG`**: Specifies the liboqs version to include.
+- **`OQSPROVIDER_TAG`**: Defines the Open Quantum Safe provider version.
+- **`INSTALLDIR`**: Sets the installation path for OQS libraries (default: `/opt/oqs`).
```

Comment:
If the default value is documented here, I'd suggest documenting it for all arguments (or none, to make future updates simpler): If you'd opt for that, what about adding "For default values, check the [Dockerfile](Dockerfile)"?